src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -72,7 +116,7 @@ public function loadChoiceList($value = null)
             return $this->choiceList;
         }
 
-        return $this->choiceList = new ArrayChoiceList($this->getTimezones(), $value);
+        return $this->choiceList = new ArrayChoiceList($this->getTimezones($this->regions), $value);
```

Comment:
> You should also avoid storing options in the form type's properties, as form types are instantiated once, it acts like a global state for the type and all its usages.

An option could be return a new default choice loader instance instead of the current one `return $this;`? e.g.:
```php
return new CallbackChoiceLoader(function () use ($options) {
    return self::getTimezones($options['region']);
});
```
But also, it involves remove the actual `ChoiceLoaderInterface` implementation and maybe I'm missing something...